keps/sig-api-machinery/1929-built-in-default/README.md

```diff
@@ -414,48 +418,73 @@ Both these cases will be handled properly, here's an example:
 type Object struct {
      // This field has an invalid zero-value, empty string is not a
      // valid string.
-     // +default=default-name
-     Name string
+     //
+     // If this is a CRD, there would be a kubebuilder specific tag, and
+     // if this is a built-in, there would be a specific validation function,
+     // but basically:
+     // +minLength=1
+     //
+     // +default="default-name"
+     Name string `json:",omitempty"`
 
      // This field is defaulted to 0 with no defaulter.
      // +default=0
      Defaulted int
 }
 ```
 
+Non-pointer fields that have a default must be omitEmpty. Kubebuilder
+and built-in defaulting generators should both prevent these from
+happening.
+
 This would generate the following OpenAPI:
 ```yaml
 Object:
   type: object
   properties:
     invalid:
       type: string
-      default: default-name
+      default: "default-name"
+      # This field would only be present for CRDs, for built-in,
+      # the empty value would be accepted and treated as missing.
+      minLength: 1
```

Comment:
Why do we want this?

Note that this is much more unnatural for ints, floats or bools. Why should strings be different?

Reply:
I'll remove it yeah